tests/unit/aggregate-step-form.spec.ts

```diff
@@ -0,0 +1,288 @@
+import { setupStore } from '@/store';
+import { Pipeline } from '@/lib/steps';
+import { VQBState } from '@/store/state';
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+
+interface ValidationError {
+  dataPath: string;
+  keyword: string;
+}
+
+describe('Aggregate Step Form', () => {
+  let emptyStore: Store<VQBState>;
+  beforeEach(() => {
+    emptyStore = setupStore({});
+  });
+
+  it('should instantiate', () => {
+    const wrapper = shallowMount(AggregateStepForm, { store: emptyStore, localVue });
+    expect(wrapper.exists()).to.be.true;
+  });
+
+  describe('WidgetMultiselect', () => {
+    it('should have exactly one WidgetMultiselect component', () => {
+      const wrapper = shallowMount(AggregateStepForm, { store: emptyStore, localVue });
+      const widgetWrappers = wrapper.findAll('widgetmultiselect-stub');
+      expect(widgetWrappers.length).to.eql(1);
+    });
+
+    it('should instantiate an WidgetMultiselect widget with proper options from the store', () => {
+      const store = setupStore({
+        dataset: {
+          headers: [{ name: 'columnA' }, { name: 'columnB' }, { name: 'columnC' }],
+          data: [],
+        },
+      });
+      const wrapper = shallowMount(AggregateStepForm, { store, localVue });
+      const widgetMultiselect = wrapper.find('widgetmultiselect-stub');
+      expect(widgetMultiselect.attributes('options')).to.equal('columnA,columnB,columnC');
+    });
+
+    it('should pass down the "on" prop to the WidgetMultiselect value prop', async () => {
+      const wrapper = shallowMount(AggregateStepForm, { store: emptyStore, localVue });
+      wrapper.setData({ editedStep: { name: 'aggregate', on: ['foo', 'bar'], aggregations: [] } });
+      await Vue.nextTick();
+      expect(wrapper.find('widgetmultiselect-stub').props().value).to.eql(['foo', 'bar']);
+    });
+
+    it('should call the setColumnMutation on input', async () => {
+      const store = emptyStore;
+      const wrapper = mount(AggregateStepForm, { store, localVue });
+      wrapper.setData({ editedStep: { name: 'aggregate', on: ['foo'], aggregations: [] } });
+      await wrapper.find(WidgetMultiselect).trigger('input');
+      expect(store.state.selectedColumns).to.eql(['foo']);
+    });
+  });
+
+  describe('WidgetList', () => {
+    it('should have exactly on WidgetList component', () => {
+      const wrapper = shallowMount(AggregateStepForm, { store: emptyStore, localVue });
+      const widgetWrappers = wrapper.findAll('widgetlist-stub');
+      expect(widgetWrappers.length).to.eql(1);
+    });
+
+    it('should pass down the "aggregations" prop to the WidgetList value prop', async () => {
+      const wrapper = shallowMount(AggregateStepForm, { store: emptyStore, localVue });
+      wrapper.setData({
+        editedStep: {
+          name: 'aggregate',
+          on: [],
+          aggregations: [{ column: 'foo', newcolumn: 'bar', aggfunction: 'sum' }],
+        },
+      });
+      await Vue.nextTick();
```

Comment:
I know there are some bad examples in our code but I think it would be better to use `await localVue.nextTick()` here, `Vue` refers to another instance (as far as I understand)